packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,244 @@
+import assert from "node:assert";
+
+import { Temporal } from "@js-temporal/polyfill";
+
+import { feature } from "../browser-compat-data/feature";
+import { Release } from "../browser-compat-data/release";
+import { browsers, highReleases, lowReleases } from "./core-browser-set";
+import { support } from "./support";
+import { Browser } from "../browser-compat-data/browser";
+import {
+  BASELINE_LOW_TO_HIGH_DURATION,
+  VERY_FAR_FUTURE_DATE,
+} from "../constants";
+import { Compat, defaultCompat } from "../browser-compat-data/compat";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: "low" | "high" | false;
+  baseline_low_date: string | null;
+  baseline_high_date: string | null;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+  compat: Compat = defaultCompat,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
```

Comment:
AFAICT there are no tests for when `checkAncestors` is true. I'm not sure when it would change the result either since BCD has lints for subfeatures not predating their parent features. Perhaps with flags that aren't cascaded the way I'd like them to be?